src/libxrpl/protocol/Feature.cpp

```diff
@@ -443,10 +443,20 @@ featureToName(uint256 const& f)
 #pragma push_macro("XRPL_ABANDON")
 #undef XRPL_ABANDON
 
+template <std::size_t N>
+constexpr auto
+enforceMaxFeatureNameSize(char const (&n)[N]) -> char const*
+{
+    static_assert(N <= maxFeatureNameSize);
+    return n;
+}
```

Comment:
Consider renaming this function to `enforceFeatureNameSize`, changing `maxFeatureNameSize = 31` to `badFeatureNameSize = 32`, and comparing `N != badFeatureNameSize`.

As discussed elsewhere, I think it's completely reasonable to block a feature name that is the same length as the hash, but I don't want to tie our hands together if we need a name longer than that in the future.

Reply:
bad grammar, fix it later I will 🙃